src/libgit2/patch_parse.c

```diff
@@ -338,23 +351,276 @@ static int parse_header_dissimilarity(
 	return 0;
 }
 
+static int check_prefix_len(
+	size_t *out_len,
+	git_patch_parsed *patch,
+	const char *path_start);
+
+static int parse_header_unquoted_pathnames_guessing_split_point(
+	char **out_old_path,
+	char **out_new_path,
+	git_patch_parsed *patch,
+	git_patch_parse_ctx *ctx)
+{
+	/*
+	 If the file hasn't been renamed, it is still possible to unambiguously
+	 parse unquoted pathnames with spaces from the header line.
+
+	 If there are two equal pathnames taking up the rest of the line, with
+	 different prefixes and separated by one whitespace, then those are
+	 definitely the pathnames.
+
+	 If the file has been renamed it's impossible to unambiguously parse the
+	 pathnames from the header's first line. But it's ok because we can still
+	 get them un-prefixed from the "rename from/to" lines.
+	 */
+	char *both_paths = NULL;
+	size_t both_paths_len = 0;
+
+	size_t offset = 0;
+	bool did_parse_paths = false;
+	bool stop_path_matching_loop = false;
+
+	if (parse_header_path_to_end_of_line(&both_paths, ctx) < 0)
+		return git_parse_err("corrupt paths in git diff header at line %"PRIuZ,
+			ctx->parse_ctx.line_num);
+
+	both_paths_len = strlen(both_paths);
+
+	/* Try to determine which space is the separator between old/new paths. */
+	for (offset = 0;
+		 offset < both_paths_len && !stop_path_matching_loop;
+		 offset++) {
+		switch (both_paths[offset]) {
+			case '\n':
+				stop_path_matching_loop = true;
+				break;
+
+			case ' ':
+			case '\t':
+			{
+				/* +/-1 for the space character itself. */
+				const char *old_path_base = both_paths;
+				const char *new_path_base = both_paths + offset + 1;
+
+				size_t old_path_len = offset;
+				size_t new_path_len = both_paths_len - old_path_len - 1;
+
+				size_t old_prefix_len = 0, new_prefix_len = 0;
+
+				size_t old_path_minus_prefix_len = 0;
+				size_t new_path_minus_prefix_len = 0;
+
+				if (check_prefix_len(&old_prefix_len,
+									 patch,
+									 old_path_base) < 0 ||
+					check_prefix_len(&new_prefix_len,
+									 patch,
+									 new_path_base) < 0)
+					continue;
+
+				old_path_minus_prefix_len = old_path_len - old_prefix_len;
+				new_path_minus_prefix_len = new_path_len - new_prefix_len;
+
+				if (old_path_minus_prefix_len == new_path_minus_prefix_len &&
+					strncmp(old_path_base + old_prefix_len,
+							new_path_base + new_prefix_len,
+							old_path_minus_prefix_len) == 0)
+				{
+					*out_old_path = strndup(old_path_base, old_path_len);
+					*out_new_path = strndup(new_path_base, new_path_len);
```

Comment:
```suggestion
					*out_old_path = git__strndup(old_path_base, old_path_len);
					*out_new_path = git__strndup(new_path_base, new_path_len);
```

I get an undefined symbol here on msvc. I think the error is dependent on the version(I'm on 2017) but at any rate, libgit2 uses it's own versions of these symbols